src/main/java/de/zalando/zmon/dataservice/data/MarkWriter.java

```diff
@@ -1,19 +1,25 @@
 package de.zalando.zmon.dataservice.data;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import de.zalando.zmon.dataservice.config.DataServiceConfigProperties;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Component;
 
 import de.zalando.zmon.dataservice.DataServiceMetrics;
 
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static de.zalando.zmon.dataservice.utils.JsonUtils.flatMapJsonNode;
+import static de.zalando.zmon.dataservice.utils.JsonUtils.flatMapJsonNumericNodes;
+import static java.util.stream.Collectors.toList;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
 @Component
 class MarkWriter implements WorkResultWriter {
```

Comment:
Having this writer effectively doubles the complexity of every check - the work required is basically identical to transforming the results for Kairos as done by the Kairos writer.